presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcDiagnosticModule.java

```diff
@@ -58,21 +58,19 @@ public void configure(Binder binder)
     @StatsCollecting
     public JdbcClient createJdbcClientWithStats(@ForBaseJdbc JdbcClient client)
     {
-        StatisticsAwareJdbcClient statisticsAwareJdbcClient = new StatisticsAwareJdbcClient(client);
```

Comment:
What was wrong?

Reply:
It looks like `StatisticsAwareJdbcClient#getStats` is not exposed in JMX so the actual stats are not exposed 

Before 
```
   Column    |  Type   | Extra | Comment
-------------+---------+-------+---------
 node        | varchar |       |
 object_name | varchar |       |
```

After 
```
                         Column                          |  Type   | Extra | Comment
---------------------------------------------------------+---------+-------+---------
 abortreadconnection.failures.fifteenminute.count        | double  |       |
 abortreadconnection.failures.fifteenminute.rate         | double  |       |
 abortreadconnection.failures.fiveminute.count           | double  |       |
```

Reply:
Can you please add a test for that?

Or even better it would be to add a trap for things like that. Do you exact reasaon why stats were not exposed? Like not allowing to use `newExporter` when bean does not expose method with `@Managed`?

Reply:
It didn't work because the `StatisticsAwareJdbcClient` instance was not a bean (never exposed at guice DI level) and so it could not get exported.

Reply:
Stats do not get exposed automatically, we expose them using `newExporter(..)...`. This code was called but `StatisticsAwareJdbcClient` was wrapped so annotated methods with `@Managed` were not accessible. It would be nice to have something to prevent such bugs.